### ISSUE TITLE
gnirehtet-run does not broadcast intent

### ISSUE THREAD

**Issue by TheDebries** [Genymobile/gnirehtet#150]
I had a connection established to my phone the other day using Gnirehtet, but now I am unable to do so. I have confirmed through `adb devices` that my phone is indeed connected but it won't work.

Whenever I try to run now, it does not show the "Broadcast intent" message.
```
D:\Program Files (HDD)\gnirehtet\gnirehtet-java>gnirehtet run
2018-11-03 19:51:14.380 I Gnirehtet: Checking gnirehtet client...
2018-11-03 19:51:14.380 I Gnirehtet: Starting relay server...
2018-11-03 19:51:14.453 I Relay: Relay server started
```
When I manually send a connection request it does appear on my phone, but no data can be transmitted.

```
D:\Program Files (HDD)\gnirehtet\gnirehtet-java>adb shell am broadcast -a com.genymobile.gnirehtet.START     -n com.genymobile.gnirehtet/.GnirehtetControlReceiver
Broadcasting: Intent { act=com.genymobile.gnirehtet.START cmp=com.genymobile.gni
rehtet/.GnirehtetControlReceiver launchParam=MultiScreenLaunchParams { mDisplayI
d=0 mFlags=0 } }
Broadcast completed: result=0
D:\Program Files (HDD)\gnirehtet\gnirehtet-java>
```
As I said above, this does create a VPN connection to my phone but no data can be transmitted, and it immediately says the VPN connection has been dropped.

NOTE: the first chunk does actually run continuously, the second one stops immediately after.
Also of interest perhaps: whenever I plug my phone into my computer via USB, it pops up twice with the action selection window.

**Comment by rom1v** [comment]
Does reinstalling on the device solve the problem?

    gnirehtet reinstall

**Comment by TheDebries** [comment]
no, it did not, neither did turning the PC on & off again nor reinstalling USB drivers nor restarting phone.

**Comment by rom1v** [comment]
Are there relevant logs in the output of `adb logcat` when you send the broadcast?

**Comment by TheDebries** [comment]
only the following:
```
11-03 20:52:14.237 15997 15997 D GnirehtetControlReceiver: Received request com.genymobile.gnirehtet.STOP
11-03 20:52:14.240 15997 15997 D GnirehtetService: Received request com.genymobile.gnirehtet.CLOSE_VPN
```
When closing the `gnirehtet run` through ^C

**Comment by rom1v** [comment]
Do you have a line `Reject to launch app`? I had this on a Asus device, and solved it by authorizing _gnirehtet_ in the "auto-start manager": https://stackoverflow.com/a/39776715/1987178

**Comment by bogotok** [comment]
same error here. running android O asus device. stuck on "Relay server started" and "broadcast completed result 0". no popup for connection request on device

**Comment by rom1v** [comment]
@bogotok, when you execute `gnirehtet run`, can you see a relevant error in the output of `adb logcat`?

**Comment by tcmaps** [comment]
> Asus device

can't believe how long i googled for that

**Comment by JamesP4** [comment]
Hey i have a little problem to 

Checking gnirehtet client...
Exception in thread "main" com.genymobile.gnirehtet.relay.CommandExecutionException: Command [adb, shell, dumpsys, package, com.genymobile.gnirehtet] returned with value 1
        at com.genymobile.gnirehtet.Main.mustInstallClient(Main.java:347)
        at com.genymobile.gnirehtet.Main.cmdStart(Main.java:250)
        at com.genymobile.gnirehtet.Main.access$600(Main.java:32)
        at com.genymobile.gnirehtet.Main$Command$6.execute(Main.java:124)
        at com.genymobile.gnirehtet.Main.main(Main.java:427)

Hhat do i need to do?

**Comment by rom1v** [comment]
@JamesP4 your device is not detected by `adb`. Did you enable USB debugging? Did you try to install [drivers](https://developer.android.com/studio/run/oem-usb.html)?

Check with `adb devices`.

And please don't reply to a random unrelated issue :wink:

**Comment by Equinoxx9** [comment]
> --------- beginning of main
08-27 15:43:49.617  4751 31113 I NearbyDirect: FastPairScanner: Same filters, scan mode and currentScan=2019/08/27 15:43:43 - 5987ms 0 results with BALANCED, no-op. [CONTEXT service_id=49 ]
08-27 15:43:49.629  4751  7872 I BeaconBle: Client requested scan, settings=BleSettings [scanMode=ZERO_POWER, callbackType=ALL_MATCHES, reportDelayMillis=0, 5 filters, 1 clients, callingClientName=Nearby]
08-27 15:43:49.652  4751  7872 I BeaconBle: Stopping scan on delegate scanner. BT state: 12
08-27 15:43:49.654  4751  7872 D BluetoothAdapter: isLeEnabled(): ON
08-27 15:43:49.658   751   751 D vendor.qti.bluetooth@1.0-ibs_handler: SerialClockVote: vote for UART CLK ON
08-27 15:43:49.658  4751  7872 I BeaconBle: Resetting - new scanner available: true
08-27 15:43:49.659   751   751 D vendor.qti.bluetooth@1.0-wake_lock: Acquire wakelock is acquired 
08-27 15:43:49.659   751   751 I vendor.qti.bluetooth@1.0-ibs_handler: DeviceWakeUp: Writing IBS_WAKE_IND
08-27 15:43:49.660   751 10682 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_ACK: 0xFC
08-27 15:43:49.660  4751  7872 I BeaconBle: 'L' hardware scan: 5 filters, scanMode=-1, reportdelay=0, callback type=1, #clients=2, workSource=WorkSource{10036 com.google.android.gms}
08-27 15:43:49.660   751 10682 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_WAKE_IND: 0xFD
08-27 15:43:49.660   751 10682 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Writing IBS_WAKE_ACK
08-27 15:43:49.663  4751  7872 I BeaconBle: Starting scan on OS scanner. BT state: 12
08-27 15:43:49.664  4751  7872 D BluetoothAdapter: isLeEnabled(): ON
08-27 15:43:49.664  4751  7872 E CheckPermission: _bluetooth code = 10 
08-27 15:43:49.667 10527 10630 D BtGatt.CustomScanStrategy: getCurrAction: 0
08-27 15:43:49.668 10527 10630 D BtGatt.CustomScanStrategy: getCurrAction: 0
08-27 15:43:49.668 10527 10694 I bt_stack: [INFO:gatt_api.cc(949)] GATT_Registerafa70725-abca-b669-e5e4-2cd27615e644
08-27 15:43:49.668 10527 10694 I bt_stack: [INFO:gatt_api.cc(969)] allocated gatt_if=4
08-27 15:43:49.670  4751  4763 D BluetoothLeScanner: onScannerRegistered() - status=0 scannerId=4 mScannerId=0
08-27 15:43:49.675 10527 10630 D BtGatt.ScanManager: isAptXLowLatencyModeEnabled: false
08-27 15:43:49.675 10527 10630 D BtGatt.CustomScanStrategy: dont startWatchWIFITraffic
08-27 15:43:49.675 10527 10630 D BtGatt.ScanManager: startWatchWIFITraffic
08-27 15:43:49.675 10527 10630 D BtGatt.CustomScanStrategy: dont resetWIFITrafficeState
08-27 15:43:49.676 10527 10694 E bt_btm  : BTM_BleObserve Observe Already Active
08-27 15:43:49.676 10527 10694 W bt_btif :  bta_dm_ble_observe BTM_BleObserve  failed. status 6
08-27 15:43:49.718   751 10682 I vendor.qti.bluetooth@1.0-ibs_handler: ProcessIbsCmd: Received IBS_SLEEP_IND: 0xFE
08-27 15:43:50.065  1707  8338 D OsAgent : Action Battery changed
08-27 15:43:50.066  1707  8338 D OsAgent : Battery::Level - 100 Scale - 100Pct - 100
08-27 15:43:50.068   778   778 V LocSvc_HIDL_IzatSubscription: [batteryLevelUpdate][772] [HS] <<<<= [HC]
08-27 15:43:50.670   748  1434 V qcbassboost: 	Effect_command: ctxt 0xe509c480, cmd 8
08-27 15:43:50.671   748  1434 V qcbassboost: 	BassBoost_getParameter(): ctxt 0xe509c480, param 0, 
08-27 15:43:50.673   748  1434 V qcbassboost: 	Effect_command: ctxt 0xe509c480, cmd 5
08-27 15:43:50.673   748  1434 V qcbassboost: 	BassBoost_setParameter(): ctxt 0xe509c480, param 1, value: 800
08-27 15:43:50.673   748  1434 V qcbassboost: 	BassBoost_setParameter() BASSBOOST_PARAM_STRENGTH value is 800
08-27 15:43:50.676   751 10690 I vendor.qti.bluetooth@1.0-ibs_handler: DeviceSleep: TX Awake, Sending SLEEP_IND
08-27 15:43:50.677   751 10690 D vendor.qti.bluetooth@1.0-ibs_handler: SerialClockVote: vote for UART CLK OFF
08-27 15:43:50.828   751 10663 D vendor.qti.bluetooth@1.0-wake_lock: Release wakelock is released 
--------- beginning of system
08-27 15:43:51.276  1707  4216 W Watchdog: !@WatchDog_1545
08-27 15:43:51.377  1068  1082 D         : power_monitor cpu0-gold-usr: open fail -1
08-27 15:43:52.052  1068  1080 D         : thermal_monitor msm_therm: open fail -1
08-27 15:43:52.053  1068  1080 D         : thermal_monitor gpu0-usr: open fail -1
08-27 15:43:52.053  1068  1080 D         : thermal_monitor cpu0-gold-usr: open fail -1
08-27 15:43:52.565  5085  5393 D TrafficStatsHelper: transmitted_validWifiData.get(0) = wlan0
08-27 15:43:54.786 19398 19427 W Settings: Setting airplane_mode_on has moved from android.provider.Settings.System to android.provider.Settings.Global, returning read-only value.
08-27 15:43:55.958  1707  8338 D OsAgent : Action Battery changed
08-27 15:43:55.959  1707  8338 D OsAgent : Battery::Level - 100 Scale - 100Pct - 100
08-27 15:43:55.966   778   778 V LocSvc_HIDL_IzatSubscription: [batteryLevelUpdate][772] [HS] <<<<= [HC]

**Comment by rom1v** [comment]
Fixed by f8681809a28a0073f791f0d9eae65da02ae73c3d.
